### PR TITLE
Support Pub/Sub Publication

### PR DESCRIPTION
**Pull request by krak3n**
# What

Adds support for Pub/Sub publication.

### CODE REVIEW

**Review by stacknowledge — APPROVED** [review]
lgtm

File: listener/pubsub_publisher.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	msg, err := json.Marshal(event)
+	if err != nil {
+		return fmt.Errorf("json marshal: %w", err)
+	}
+
+	result := p.topic.Publish(context.Background(), &pubsub.Message{
+		Data: msg,
+	})
+
+	<-result.Ready()
```
Comment (aranw) [inline listener/pubsub_publisher.go]: Unnecessary call to `<-result.Ready()` as it is checked in `result.Get(ctx)` 
```suggestion
```

File: listener/pubsub_publisher.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+	_, err = result.Get(ctx)
+	if err != nil {
+		return fmt.Errorf("pubsub publish: %w", err)
+	}
```
Comment (aranw) [inline listener/pubsub_publisher.go]:
```suggestion
	if _, err := result.Get(ctx); err != nil {
		return fmt.Errorf("pubsub publish: %w", err)
	}
```

**Review by aranw — COMMENTED** [review]
Few changes I think mostly looks good though

**Pull request merged**